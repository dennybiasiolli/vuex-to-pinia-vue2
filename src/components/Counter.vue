<template>
  <div>
    <div>
      count is: {{ count }}
    </div>
    <div>
      isEven is: {{ isEven }}
      |
      isOdd is: {{ isOdd }}
    </div>
    <button @click="increment">increment sync</button>
    &nbsp;
    <button @click="incrementAsync">increment async</button>
  </div>
</template>

<script>
import { mapState, mapActions } from 'pinia'
import { useStore } from '@/stores/main'

export default {
  computed: {
    ...mapState(useStore, ['count', 'isEven', 'isOdd']),
  },
  methods: {
    ...mapActions(useStore, ['increment', 'incrementAsync']),
  },
}
</script>
