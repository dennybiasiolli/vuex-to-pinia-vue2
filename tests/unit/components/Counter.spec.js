import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Counter from '@/components/Counter.vue'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('Counter.vue', () => {
  let store;
  let mutations = {
    increment: jest.fn(),
  }
  let actions = {
    incrementAsync: jest.fn(),
  }
  beforeEach(() => {
    store = new Vuex.Store({
      state: {
        count: 0,
      },
      getters: {
        isEven: () => true,
      },
      mutations,
      actions,
    })
  })
  test('should respect the snapshot', () => {
    const wrapper = shallowMount(Counter, { localVue, store })
    expect(wrapper.element).toMatchSnapshot()
  })

  test('increment button should call store mutation', () => {
    const wrapper = shallowMount(Counter, { localVue, store })
    wrapper.findAll('button').at(0).trigger('click')
    expect(mutations.increment).toHaveBeenCalled()
  })

  test('incrementAsync button should call store action', () => {
    const wrapper = shallowMount(Counter, { localVue, store })
    wrapper.findAll('button').at(1).trigger('click')
    expect(actions.incrementAsync).toHaveBeenCalled()
  })
})
