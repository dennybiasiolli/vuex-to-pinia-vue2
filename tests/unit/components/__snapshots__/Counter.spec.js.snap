// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter.vue should respect the snapshot 1`] = `
<div>
  <div>
    
    count is: 0
  
  </div>
   
  <div>
    <span>
       is even
    </span>
  </div>
   
  <button>
    increment sync
  </button>
  
   
  
  <button>
    increment async
  </button>
</div>
`;
